mycroft/gui/__init__.py

```diff
@@ -186,13 +189,19 @@ def _pages2uri(self, page_names):
         page_urls = []
         for name in page_names:
             if name.startswith("SYSTEM"):
-                page = resolve_resource_file(join('ui', name))
+                if self.serving_http:
+                    page = f"{self.remote_url}/system/ui/{name}"
+                else:
+                    page = resolve_resource_file(join('ui', name))
             else:
                 page = self.skill.find_resource(name, 'ui')
+                if self.serving_http:
+                    page = page.replace(self.base_skill_dir,
+                                        join(self.remote_url, "skills"))
             if page:
                 if self.remote_url:
                     page_urls.append(self.remote_url + "/" + page)
-                elif page.startswith("file://"):
+                elif "://" in page:
```

Comment:
If `self.remote_url` is defined, this clause will never be parsed. I added this because I think `://` will always relate to an absolute path, so some absolute path like `smb://` or `http://` could be defined without `self.remote_url` (i.e. I want to reference an exact QML file I host without leaving it up to a file resolver method)

Reply:
Could it still run the startswith or would that be some slow regex stuff?